spec/models_spec.rb

```diff
@@ -0,0 +1,141 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  let!(:user) { User.new(id: 1, name: 'Foo Bar', email: 'foo@bar.com', password: '12345678') }
+
+  describe 'Associations' do
+    it { should have_many(:comments) }
+    it { should have_many(:likes) }
+    it { should have_many(:posts) }
+    it { should have_many(:friendships) }
+    it { should have_many(:inverse_friendships) }
+  end
+
+  describe 'Validations' do
+    it 'Is valid with name.' do
+      expect(user).to be_valid
+    end
+
+    it 'Name must be 20 characters or less.' do
+      user.name = 'Joao Paulo Dias Franca'
+      expect(user).to_not be_valid
+    end
+
+    it 'Name must be present.' do
+      user.name = nil
+      expect(user).to_not be_valid
+    end
+  end
+end
+
+RSpec.describe Post, type: :model do
+  let!(:user) { User.new(id: 1, name: 'Foo Bar', email: 'foo@bar.com', password: '12345678') }
+  let!(:post) { Post.new(content: 'Hello World!', user_id: 1) }
+
+  describe 'Associations' do
+    it { should have_many(:comments) }
+    it { should have_many(:likes) }
+  end
+
+  describe 'Validations' do
+    it 'Is valid with content.' do
+      user.save
+      expect(post).to be_valid
+    end
+
+    it 'Content must be 1000 characters or less.' do
+      post.content = 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
+      expect(post).to_not be_valid
```

Comment:
You save some lines by simple using `times` method to generate your 1000 charater string
```ruby
str = ' '
1000.times { str << 'a' }
post.content = str
```